presto-spi/src/main/java/com/facebook/presto/spi/Page.java

```diff
@@ -118,11 +118,10 @@ public void compact()
             if (block instanceof DictionaryBlock) {
                 continue;
             }
-            if (block.getSizeInBytes() < block.getRetainedSizeInBytes()) {
-                // Copy the block to compact its size
-                Block compactedBlock = block.copyRegion(0, block.getPositionCount());
-                blocks[i] = compactedBlock;
-            }
+
+            // Copy the block to compact its size
+            Block compactedBlock = block.copyRegion(0, block.getPositionCount());
+            blocks[i] = compactedBlock;
```

Comment:
inline `compactedBlock`

Reply:
This belong to the commit `Remove the unnecessary heuristic before compacting block`